stdlib/http/src/main/java/org/ballerinalang/net/http/websocket/client/listener/ClientHandshakeListener.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.net.http.websocket.client.listener;
+
+import org.ballerinalang.jvm.values.ObjectValue;
+import org.ballerinalang.net.http.websocket.WebSocketResourceDispatcher;
+import org.ballerinalang.net.http.websocket.WebSocketUtil;
+import org.ballerinalang.net.http.websocket.server.WebSocketConnectionInfo;
+import org.wso2.transport.http.netty.contract.websocket.WebSocketConnection;
+import org.wso2.transport.http.netty.message.HttpCarbonResponse;
+
+/**
+ * Handshake listener of WebSocket client.
+ *
+ * @since 1.2.0
+ */
+public class ClientHandshakeListener implements HandshakeListener {
```

Comment:
Do we need this class base class should be WebSocketHandshakeListener right

Reply:
Yes as the dispatchOnError() used to close the connection between client and server(terminate the connector thread) but WebSocketHandshakeListener is a base class for RetryHandshakeListener and FailoverHandshakeListener. Therefore we can't call that dispatchOnError() in the base class. So, I implemented this class to handle dispatchOnError() for normal webSocketClient.